README.md

```diff
@@ -1,4 +1,140 @@
-# MicroCluster
+# Microcluster
 ## Introduction
-MicroCluster is a helper package for bootstrapping and managing a dqlite cluster.
-Based on [go-dqlite](https://github.com/canonical/go-dqlite).
+
+Microcluster is a Go library that provides an easy to use framework for creating and managing highly available [dqlite](https://github.com/canonical/dqlite) clusters. It provides an extensible API and database which can be used to directly connect to your underlying services.
+
+Extend Microcluster with your own API endpoints, schema and API extensions, lifecycle actions, and additional network listeners.
+
+## Import Microcluster
+
+Get the latest LTS release of Microcluster:
+
+```
+$ go get github.com/canonical/microcluster/v2@latest
+```
+
+## Usage
+For a step-by-step practical example of creating a Microcluster project, check out the [example package](example). Examine the code or build and test it out with the included Makefile.
+
+All of Microcluster's state is stored in the `StateDir` directory. Define a new Microcluster app:
+
+```go
+m, err := microcluster.App(microcluster.Args{StateDir: "/path/to/state"})
+if err != nil {
+    // ...
+}
+```
+
+Extensions to Microcluster must be defined in `DaemonArgs`:
+```go
+
+dargs := microcluster.DaemonArgs{
+    Version: "0.0.1",
+
+    ExtensionsSchema: []schema.Update{
+        // Sequential list of modifications to the database.
+    },
+
+    APIExtensions: []string{
+        // Sequential list of API capabilities.
+    },
+
+    Servers: map[string]rest.Server{
+        "my-server": {
+            CoreAPI: true,
+            ServeUnix: true,
+            Resources: []rest.Resources{
+                {
+                    PathPrefix: "1.0",
+                    Endpoints: []rest.Endpoint{
+                        // Additional API endpoints to offer on "/1.0".
+                    },
+                },
+            }
+        }
+    }
+}
+```
+
+Start the daemon with your configuration:
+```go
+err := m.Start(ctx, dargs)
+if err != nil {
+    // ...
+}
+```
+
+### Define lifecycle actions (Hooks):
```

Comment:
"Where can I go to see all the hooks Microcluster provides?"

Maybe link [here](https://pkg.go.dev/github.com/canonical/microcluster@v0.0.0-20240731145923-49573627997a/internal/state#Hooks)?

Reply:
Linked to the code in the repo since it links to the `v3` branch head instead of a specific commit.